### ISSUE TITLE
Fen not letting me create more than one shorcut folder

### ISSUE THREAD

**Issue by ivarbrandt** [Tikipeter/repository.tikipeter#724]
After the updates, it seems that Fen won't let me create more than one shortcut folder in the root menu.

When I try to create one for the first time, it shows the dialog box "create a new shortcut folder", I click OK, name it, and it creates it just fine. However when I go to create another one it skips the option to create a new one and just creates a duplicate of the previous shortcut folder that I just created. The only option it gives me is where to place the duplicate. I also tried to add a menu item to a shortcut folder to see if it would let me create a new one but it doesn't work there either.

I thought that it could be because I just installed the updates over the old version so I tried wiping the add on and still had the same issue. Finally I did a fresh install of all components of 
Fen including My Accounts and Fenomscrapers, tried again and the behavior persisted.

I've attached my kodi log.

[kodi.log](https://github.com/Tikipeter/repository.tikipeter/files/8252925/kodi.log)

**Comment by Tikipeter** [comment]
![screenshot00001](https://user-images.githubusercontent.com/28930002/158404507-4258d038-5235-406b-8843-883fbbf23509.png)

Unfortunately it is working OK for me. I just made these 2 test shortcut folders.

Go to userdata/addon_data/plugin.video.fen and make sure there is a `databases` folder. Inside that folder there should be 9x `.db` files, including a `navigator.db` file. Make sure all those files are over 0kb in size. If any are 0kb in size, delete them.

Have you made any other changes to the Fen menus that have applied and persisted correctly before this?
It sounds like maybe that `navigator.py` file is corrupted or was not made correctly in the first place.

**Comment by ivarbrandt** [comment]
I checked and all the db files are over 0kb in size.
I had lots of custom menus and shortcut folders before the new updates and everything was working well.

Ahh I didn't try that way! It's working now! Thanks man.

One more thing before you close this please, my Next Episodes isn't loading for some reason.
Here is the log.

[kodi.log](https://github.com/Tikipeter/repository.tikipeter/files/8254704/kodi.log)

**Comment by Tikipeter** [comment]
Again mine is working!
Let me know/screenshot your Next Episodes settings as you may have something different there that is triggering the failure.

If it's not working from a widget, make sure to check if it works from inside the addon. That can narrow down the issue also.

**Comment by ivarbrandt** [comment]
Lol I'm having all sorts of issues today. So I went back into Fen and got this error "Connection aborted, connection reset error". It was related to Trakt. 

I access everything from inside Fen directly. I don't use widgets or skins.

Here's a screenshot of my Next Episodes setting.

![Screenshot_20220315-094041](https://user-images.githubusercontent.com/54084201/158416320-eacc34e8-8440-4416-b897-ac8396ba65ee.png)

**Comment by ivarbrandt** [comment]
Imma try something. I think I know what's going on.

**Comment by Tikipeter** [comment]
Well that looks like a nasty problem I haven't seen before.

It looks like there is some funky data coming in from trakt. Possibly no relevant data at all.

First check if any relevant data from trakt is being used i.e. watched statuses of movies and tv shows. Looks like the call for trakt recent activities on startup is failing completely. Also see if you can navigate successfully into any of your Trakt lists to see if you get any successful connection to trakt at all.

Easiest thing to do here would be to delete the `traktcache.db` then immediately shut Kodi down and then restart it.

Next easiest is to post here a copy of your `trakt.db` file so I can see what data trakt is returning for certain things.

Failing that I will have to post here a test version with some logging so that I can see what data is being returned from Trakt.

**Comment by Tikipeter** [comment]
Hang on, I just saw this in your log..
`TraktMonitor Service Update FEN TraktMonitor - Aborted. No Trakt Account Active - Next Update in 30 minutes...`

Fen doesn't think you have a Trakt account authed.
Be sure to run a sync on My Accounts settings to Fen. Then be sure to hit OK to save the setting. Then go back in and make sure you have a valid entry in the Username field in Fen settings for Trakt.

**Comment by ivarbrandt** [comment]
Yeah it's strange because movies are working but anything that has to do with shows and Trakt is failing completely.

Where can I find the trakt.db file? I don't see it in the databases folder in plugin.video.fen.

Wooow that's strange cause I definitely have my Trakt account authed, otherwise I wouldn't be able to access my movies.

**Comment by Tikipeter** [comment]
>Where can I find the trakt.db file? I don't see it in the databases folder in plugin.video.fen.

It needs to be there. If it's not, restart Kodi and it should be made upon restart. There's something funky going on with your install if it isn't there.

**Comment by Tikipeter** [comment]
Make sure Openwizard is not installed.

**Comment by Tikipeter** [comment]
Also make sure the username for trakt in My Accounts matches the username for Trakt in Fen's settings.

**Comment by ivarbrandt** [comment]
Openwizard isn't installed. I did have it installed previously though. Not on this fresh install tho. Literally the only add on I have on this is Fen lol.

Also the usernames match in My Accounts and Fen settings.

**Comment by Tikipeter** [comment]
Has this been going on only very recently after the last update to Fen (v.2.6.08)?
If so, install version 2.6.07 from here: https://github.com/Tikipeter/repository.tikipeter/tree/master/zip/plugin.video.fen
Restart Kodi and check if the 'traktcache.db' is made correctly on that version.

**Comment by ivarbrandt** [comment]
Ok I will try that. 
Apparently I missed quite a few updates as I updated directly from 2.5.21.

I tried that version (2.6.07) and it's having the same problem and throwing the same error.

Later when I get a chance I'll go back and update past versions one by one and see which one it broke on.

**Comment by meesterexx** [comment]
I would like to add that I have seen issues with Trakt cache for a few months, I want to say since January, but it may have been before, I can't remember. I had to stop clearing the Trakt cache from my build on my dev box before I copied it to my Shield. 

What happens if I forget and clear Trakt cache, next time I load Kodi, my next epsiodes widget never loads and next episodes from inside the addon won't load at that point either.

What I have to do to correct it (if I don't have a backup db file) is start kodi and quickly go into the addon and go to next episodes before the widget code has a chance to run. Then it will load in the addon. I have to exit kodi at that point and restart kodi to get the widget to work, reload skin won't work. This is on multiple skins, so I don't think it is specifically a skin issue. Also of note, I usully have reuse language invoker set to false in the addon because most skins bomb after playback stops if launched from a widget. (Maybe not related, but I wanted you to know the conditions it was occurring under).

So this post made me curious, I started from a completely empty kodi 19.4 and installed Fen. After authorizing trakt in my accounts, it created traktcache.db at 28k, but the tables were empty:

![progress](https://user-images.githubusercontent.com/13154733/158466531-f6ecf875-092c-4db5-aece-ba3538142057.png)

![trakt_data](https://user-images.githubusercontent.com/13154733/158466577-b4ee855f-2df2-4f0b-81aa-824e289ac0f6.png)

![watched_status](https://user-images.githubusercontent.com/13154733/158466592-aa21b4e7-2a64-4b37-b189-28185b60cfec.png)

I exited and restarted kodi and at that point and all tables populated as expected:
![after-exit](https://user-images.githubusercontent.com/13154733/158466686-c77f8242-ca16-4a6c-9fc3-a603c46b7654.png)

I don't know if any of this is helpful, but I wanted to relate my experiences with the issue so you could see it's not completely isolated. Please let me know if there's anything I can do to help or test in anyway. Fen is my go to and I appreciate all the work that you have put into it.

**Comment by ivarbrandt** [comment]
So I downloaded all the past versions from 2.6.01 to 2.6.07 and all of them threw an error. Idk if the same error but here's the log.


[kodi.log](https://github.com/Tikipeter/repository.tikipeter/files/8257093/kodi.log)

**Comment by Tikipeter** [comment]
It's still saying you don't have an active trakt account.

Can you get a screenshot of the contents of your `databases` folder in userdata?

**Comment by ivarbrandt** [comment]
> It's still saying you don't have an active trakt account.
> 
> Can you get a screenshot of the contents of your `databases` folder in userdata?

![Screenshot_20220315-221653](https://user-images.githubusercontent.com/54084201/158515845-cd3b6a18-0991-4a71-ab6a-f2738774b18e.png)

**Comment by ivarbrandt** [comment]
Well that's a good sign. Thanks for continuing to look into it.

**Comment by derrgoo** [comment]
Seems like everyone would be having this issue but my trakt and progress working perfect. I find also registering watched is faster as well. 

Haven't seen any of the above issues. 🤷‍♀️
